src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -184,11 +195,47 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
         const cancelToken = new CancellationTokenSource();
         this.wasPythonInstalledWhenFetchingControllers = this.extensionChecker.isPythonExtensionInstalled;
         const connections = await this.getKernelConnectionMetadata(listLocalNonPythonKernels, cancelToken.token);
-
         // Now create the actual controllers from our connections
         this.createNotebookControllers(connections);
+        // If we're listing Python kernels & there aren't any, then add a placeholder for `Python` which will prompt users to install python
+        if (!listLocalNonPythonKernels) {
+            if (connections.some((item) => isPythonKernelConnection(item))) {
+                this.removeNoPythonControllers();
+            } else {
+                this.regsiterNoPythonControllers();
+            }
+        }
     }
+    private removeNoPythonControllers() {
+        this.notbeookNoPythonController?.dispose();
```

Comment:
Does it get updated after the python extension is installed/interpreters are found?

Reply:
Yes, existing code (if you don't have python installed & then install it, we re-query list of kernels)